### ISSUE TITLE
Add `.jsx` support for `vue/html-self-closing`

### ISSUE THREAD

**Issue by wenfangdu** [vuejs/eslint-plugin-vue#1507]
**What rule do you want to change?**

`vue/html-self-closing`

**Does this change cause the rule to produce more or fewer warnings?**

More warnings.

**How will the change be implemented? (New option, new default behavior, etc.)?**

Currently, the following rule won't affect `.jsx` files:

```json
{
  "vue/html-self-closing": [
    "error",
    {
      "html": {
        "component": "always",
        "normal": "always",
        "void": "always"
      }
    }
  ]
}
```

For achieving the same linting, we have to resort to using [`eslint-plugin-react`](https://github.com/yannickcr/eslint-plugin-react), which is quite awkward given we're writing Vue's JSX, would be great if `eslint-plugin-vue` natively supports it.

**Please provide some example code that this change will affect:**

```jsx
import { defineComponent } from 'vue'

export default defineComponent({
  setup() {
    return () => <VTable></VTable>
  },
})
```

**What does the rule currently do for this code?**

No effect.

**What will the rule do after it's changed?**

Warns `VTable` should be self-closed.

**Additional context**

Although currently, I'm proposing adding `.jsx` support for one rule, but I think this issue should be generalized, there are many other rules that don't apply to `.jsx` files where they are expected to.

**Comment by ota-meshi** [comment]
If there is already a rule in `eslint-plugin-react` for checking JSX, I think it's a good idea to use that rule.
If we create rules that do the same check in different plugins, the rules can conflict with each other and confuse the user.
